packages/mermaid/src/diagrams/class/classDb.ts

```diff
@@ -24,713 +24,720 @@ import type {
-    classRelation.id1 = `interface${interfaces.length - 1}`;
-  } else if (
-    classRelation.relation.type2 === relationType.LOLLIPOP &&
-    !invalidTypes.includes(classRelation.relation.type1)
-  ) {
-    addClass(classRelation.id1);
-    addInterface(classRelation.id2, classRelation.id1);
-    classRelation.id2 = `interface${interfaces.length - 1}`;
-  } else {
-    addClass(classRelation.id1);
-    addClass(classRelation.id2);
-  }
-
-  classRelation.id1 = splitClassNameAndType(classRelation.id1).className;
-  classRelation.id2 = splitClassNameAndType(classRelation.id2).className;
-
-  classRelation.relationTitle1 = common.sanitizeText(
-    classRelation.relationTitle1.trim(),
-    getConfig()
-  );
-
-  classRelation.relationTitle2 = common.sanitizeText(
-    classRelation.relationTitle2.trim(),
-    getConfig()
-  );
-
-  relations.push(classRelation);
-};
-
-/**
- * Adds an annotation to the specified class Annotations mark special properties of the given type
- * (like 'interface' or 'service')
- *
- * @param className - The class name
- * @param annotation - The name of the annotation without any brackets
- * @public
- */
-export const addAnnotation = function (className: string, annotation: string) {
-  const validatedClassName = splitClassNameAndType(className).className;
-  classes.get(validatedClassName)!.annotations.push(annotation);
-};
-
-/**
- * Adds a member to the specified class
- *
- * @param className - The class name
- * @param member - The full name of the member. If the member is enclosed in `<<brackets>>` it is
- *   treated as an annotation If the member is ending with a closing bracket ) it is treated as a
- *   method Otherwise the member will be treated as a normal property
- * @public
- */
-export const addMember = function (className: string, member: string) {
-  addClass(className);
-
-  const validatedClassName = splitClassNameAndType(className).className;
-  const theClass = classes.get(validatedClassName)!;
-
-  if (typeof member === 'string') {
-    // Member can contain white spaces, we trim them out
-    const memberString = member.trim();
-
-    if (memberString.startsWith('<<') && memberString.endsWith('>>')) {
-      // its an annotation
-      theClass.annotations.push(sanitizeText(memberString.substring(2, memberString.length - 2)));
-    } else if (memberString.indexOf(')') > 0) {
-      //its a method
-      theClass.methods.push(new ClassMember(memberString, 'method'));
-    } else if (memberString) {
-      theClass.members.push(new ClassMember(memberString, 'attribute'));
+export class ClassDB implements DiagramDB {
+  private relations: ClassRelation[] = [];
+  private classes = new Map<string, ClassNode>();
+  private readonly styleClasses = new Map<string, StyleClass>();
+  private notes: ClassNote[] = [];
+  private interfaces: Interface[] = [];
+  // private static classCounter = 0;
+  private namespaces = new Map<string, NamespaceNode>();
+  private namespaceCounter = 0;
```

Comment:
Should this be out of the class like `classCounter` is? It seems to be used to create global DOM IDs.